addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -408,15 +421,48 @@ def _import_fattura_pa(self, tree, invoice):
 
             # Type must be present in the context to get the right behavior of the _default_journal method (account.move).
             # journal_id must be present in the context to get the right behavior of the _default_account method (account.move.line).
-            elements = tree.xpath('//CessionarioCommittente//IdCodice')
-            company = elements and self.env['res.company'].search([('vat', 'ilike', elements[0].text)], limit=1)
-            if not company:
-                elements = tree.xpath('//CessionarioCommittente//CodiceFiscale')
-                company = elements and self.env['res.company'].search([('l10n_it_codice_fiscale', 'ilike', elements[0].text)], limit=1)
+
+            default_move_type = self.env.context.get('default_move_type')
+            # The default_move_type is ...
+            #   * ... set when clicking on the "Upload" button on the invoices or bills views.
+            #     In that case we want to only import it as invoice or bill resepectively
+            #   * ... not set when importing from cron
+            if default_move_type is None:
+                # Try importing as incoming or outgoing move
+                incoming_possibilities = [True, False]
+            elif default_move_type in invoice.get_purchase_types(include_receipts=True):
+                # Only try importing as incoming move
+                incoming_possibilities = [True]
+            elif default_move_type in invoice.get_sale_types(include_receipts=True):
+                # Only try importing as outgoing move
+                incoming_possibilities = [False]
+            else:
+                _logger.warning("Cannot handle default_move_type '%s'.", default_move_type)
+                continue
+
+            # Consider the import via the "Upload" button on the invoices or bills view.
+            # There we only want to import moves for the currently allowed companies.
```

Comment:
I'd like to remove all comments in this section and make just one both simpler and more exhaustive.

What do you think?
```
            # Two cases to determine whether it's incoming (bill) or outgoing (invoice):
            # - Import from Tax Agency (cron).
            #   No `default_move_type` is found in the context.
            #   We look for VAT in both the Seller and the Buyer XML sections 
            #   to determine the Company and the Document Type.
            # - User clicks the "IMPORT" button
            #   Correct `default_move_type` is present in the context.
            #   The company must be one of the allowed ones for the current user.
            #   Import button on "Invoices" view: VAT must be in the Seller section.                
            #   Import button on "Vendor Bills" view: VAT must be in the Buyer section.
            default_move_type = self.env.context.get('default_move_type')
            if default_move_type is None:
                incoming_possibilities = [True, False]
            elif default_move_type in invoice.get_purchase_types(include_receipts=True):
                incoming_possibilities = [True]
            elif default_move_type in invoice.get_sale_types(include_receipts=True):
                incoming_possibilities = [False]
```

Reply:
Makes sense. I edited the comment text a bit. Is it okay?

Reply:
I moved / modified the comment a bit
https://github.com/odoo/odoo/compare/e8f9c711d8bc7d348a49f8c1c7053f06d46f4d02..5fc0b375a205b4763fbe334ba56343007782972f
Or should I rather remove it completely?

More detailed (15 specific) explanation of the original comment (imo).
```python
            # Setup the context for the Invoice Form
            # The 'default_move_type' i.e. determines the `move_type` and `journal_id` (see `_get_default_journal` of model 'account.move').
            # In the account move form view the 'default_move_type' and 'journal_id' are be passed to the context of the 'invoice_line_ids' field.
            # These values are later used to determine the `account_id` of the created invoice lines (see `default_get` of model 'account.move.ine').
```

Reply:
What we added before, is a (long but necessary) comment explaining why the code there goes great length to determine whether we are in an inbound or outbound situation, and why we come up with checking the context to determine things.

This down below instead, is just general confusing documentation on how the move gets created, and it's actually Odoo's standard way with nothing special, and it's done in `account_edi`, not here.

Just feel free to remove it as it has no use here, it's just added confusion.